src/blog/converting-svg-to-image-png-webp-etc-in-javascript.md

```diff
@@ -0,0 +1,337 @@
+
+```javascript
+export const fetchImage = async ({imgSrc}: {imgSrc: string}): Promise<string | undefined> => {
+  const data: Response = await fetch(imgSrc);
+  const blob: Blob = await data.blob();
+
+  const base64: string = await toBase64({blob});
+
+  return base64.split(',')?.[1];
+};
+
+const toBase64 = ({blob}: {blob: Blob}): Promise<string> => {
+  return new Promise<string>((resolve, reject) => {
+    try {
+      const reader: FileReader = new FileReader();
+      reader.onloadend = () => {
+        const {result} = reader;
+        resolve(result as string);
+      };
+
+      reader.readAsDataURL(blob);
+    } catch (err) {
+      reject(err);
+    }
+  });
+};
+```
+
+In above code snippet, the `imgSrc` is the URL to the image, the logo, that should be embedded. It is first fetched, then transformed to a `blob` and finally converted to a `base64` string.
+
+*****
+
+### Convert To Image
+
+Basically, the conversion process happens in two steps:
+
+* SVG to Canvas
+* Canvas to Image (Blob)
+
+Translated to code, these steps can be chained in a function.
+
+```javascript
+@Method()
+async toBlob(type: string = 'image/webp'): Promise<Blob> {
+  const canvas: HTMLCanvasElement = 
+        await svgToCanvas({svg: this.svgRef});
+  return canvasToBlob({canvas, type});
+}
+```
+
+As you may notice, the above method define a default mime type for the export `image/webp`. According to my tests, it does also the job for other format such as `image/png` and `image/jpg`.
+
+*****
+
+#### SVG To Canvas
+
+In one of my previous work, a [Figma plugin](https://github.com/deckgo/figma-deckdeckgo-plugin), I already developed a function that convert SVG to `HTMLCanvasElement`.
+
+```javascript
+export const transformCanvas = ({index}: Frame): Promise<SvgToCanvas | undefined> => {
+  return new Promise<SvgToCanvas | undefined>((resolve) => {
+    const svg: SVGGraphicsElement | null =
+      document.querySelector(`div[frame="${index}"] svg`);
+
+    if (!svg) {
+      resolve(undefined);
+      return;
+    }
+
+    const {width, height} = svgSize(svg);
+
+    const blob: Blob =
+      new Blob([svg.outerHTML], 
+              {type: 'image/svg+xml;charset=utf-8'});
+    const blobURL: string = URL.createObjectURL(blob);
+
+    const image = new Image();
+
+    image.onload = () => {
+      const canvas: HTMLCanvasElement = 
+                    document.createElement('canvas');
+
+      canvas.width = width;
+      canvas.height = height;
+
+      const context: CanvasRenderingContext2D | null =
+        canvas.getContext('2d');
+      context?.drawImage(image, 0, 0, width, height);
+
+      URL.revokeObjectURL(blobURL);
+
+      resolve({
+        canvas,
+        index
+      });
+    };
+
+    image.src = blobURL;
+  });
+};
+```
+
+At first, I had the feeling it would be a piece of cake to re-implement the exact same above function. Unfortunately, “feeling” was the only valid keyword in such sentence 😂.
```

Comment:
```suggestion
At first, I had the feeling it would be a piece of cake to re-implement the exact same above function. Unfortunately, “feeling” was the only valid keyword in that sentence 😂.
```